specification.md

```diff
@@ -1855,6 +1860,20 @@ for this connection.
 This field is optional: if omitted the client SHOULD NOT use a client-side certificate.
 This field can be used to perform a client certificate revocation/rotation.
 
+##### OpAMPConnectionSettings.heartbeat_interval_seconds
+
+The Client should use the offered heartbeat interval to periodically send an AgentToServer
+message. At a minimum the instance_uid field MUST be set. It is recommended that the Agent
+also set ComponentHealth as well. An HTTP based-client MUST use the heartbeat interval as
+its polling interval.
+
+A heartbeat is used to keep a load balancer connection active and inform the server that
+the Agent is still alive and active. A server could use the heartbeat to make decisions about
+the liveness of the connected Agent.
```

Comment:
This is a breaking change as it's written, right? Agents pre-heartbeat will ignore the heartbeat interval, the server will determine they aren't alive, then (presumably) disconnect the agent.

I think this could easily be made backwards compatible if the agent explicitly indicated that it supports this e.g. through it's capabilities.

Reply:
maybe not, there's guidance [here](https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#ignorable-capability-extensions). I think given we also want to use this to control the HTTP polling interval this may be a non-ignorable capability extension and would therefore require a new AgentCapability. Is there anyway around that @tigrannajaryan @andykellr ?

Reply:
I think a new flag for `AgentCapabilities_ReportsHeartbeat` would be appropriate so that servers know if they should expect heartbeats.